cpp/examples/arrow/compute_and_write_csv_example.cc

```diff
@@ -0,0 +1,140 @@
+// ./comparison_example file:///$PWD
+
+int main(int argc, char** argv) {
+  if (argc < 2) {
+    std::cout << "Please enter the path to which you want data saved" << std::endl;
+    // Fake success for CI purposes.
+    return EXIT_SUCCESS;
+  }
+
+  // Make Arrays
+  arrow::NumericBuilder<arrow::Int64Type> int64_builder;
+  arrow::BooleanBuilder boolean_builder;
+
+  // Make place for 8 values in total
+  ARROW_RETURN_NOT_OK(int64_builder.Resize(8));
+  ARROW_RETURN_NOT_OK(boolean_builder.Resize(8));
+
+  // Bulk append the given values
+  std::vector<int64_t> int64_values = {1, 2, 3, 4, 5, 6, 7, 8};
+  ARROW_RETURN_NOT_OK(int64_builder.AppendValues(int64_values));
+  std::shared_ptr<arrow::Array> array_a;
+  ARROW_RETURN_NOT_OK(int64_builder.Finish(&array_a));
+  int64_builder.Reset();
+  int64_values = {2, 5, 1, 3, 6, 2, 7, 4};
+  std::shared_ptr<arrow::Array> array_b;
+  ARROW_RETURN_NOT_OK(int64_builder.AppendValues(int64_values));
+  ARROW_RETURN_NOT_OK(int64_builder.Finish(&array_b));
+
+  // Cast the arrays to their actual types
+  auto int64_array_a = std::static_pointer_cast<arrow::Int64Array>(array_a);
+  auto int64_array_b = std::static_pointer_cast<arrow::Int64Array>(array_b);
+  // Explicit comparison using a loop
+  for (int64_t i = 0; i < 8; i++) {
+    if ((!int64_array_a->IsNull(i)) & (!int64_array_b->IsNull(i))) {
+      bool comparison_result = int64_array_a->Value(i) > int64_array_b->Value(i);
+      boolean_builder.UnsafeAppend(comparison_result);
+    }
+  }
+  std::shared_ptr<arrow::Array> array_c;
+  ARROW_RETURN_NOT_OK(boolean_builder.Finish(&array_c));
+  std::cout << "Array explicitly compared" << std::endl;
+
+  // Try a compute function for comparison
+  arrow::Datum compared_datum;
+  std::shared_ptr<arrow::Array> array_d;
+  arrow::Result<arrow::Datum> st_compared_datum =
+      arrow::compute::CallFunction("greater", {array_a, array_b});
+  if (st_compared_datum.ok()) {
+    compared_datum = std::move(st_compared_datum).ValueOrDie();
+    array_d = compared_datum.make_array();
+  } else {
+    std::cerr << st_compared_datum.status() << std::endl;
+  }
+  std::cout << "Arrays compared using a compute function" << std::endl;
+  // Create a table
+  auto schema =
+      arrow::schema({arrow::field("a", arrow::int64()), 
+                     arrow::field("b", arrow::int64()),
+                     arrow::field("a>b? (self written)", arrow::boolean()),
+                     arrow::field("a>b? (arrow)", arrow::boolean())});
+  std::shared_ptr<arrow::Table> my_table =
+      arrow::Table::Make(schema, {array_a, array_b, array_c, array_d});
+
+  std::cout << "Table created" << std::endl;
+
+  // Create a folder to output the data
+  std::string uri = argv[1];
+  std::string root_path;
+  auto fs = arrow::fs::FileSystemFromUri(uri, &root_path).ValueOrDie();
+  std::string base_path = root_path + "/csv_dataset";
+  std::cout << "Base path " << base_path << std::endl;
+  ARROW_RETURN_NOT_OK(fs->CreateDir(base_path));
```

Comment:
appveyor ci error needs be fixed:
https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/40927445/job/rmy2wlpipu4xg6ny#L1201

Reply:
@cyb70289 Appveyor build passes. Tests in C++ / AMD64 MacOS 10.15 C++  pass, but then it seems to time out.

Reply:
@cyb70289 All tests pass.

Reply:
@bkmgit , I'm in holiday next week, may not have time to review your code. Will do when I'm back asap.

Reply:
ok